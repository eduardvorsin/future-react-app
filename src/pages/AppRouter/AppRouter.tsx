import React from 'react';
import {
  createBrowserRouter,
  RouterProvider,
} from 'react-router-dom';
import BookDescription from '../../components/BL/BookDescription/BookDescription';

import HomePage from '../HomePage/HomePage';
import NotFoundPage from '../NotFoundPage/NotFoundPage';

const router = createBrowserRouter(
  [
    {
      path: '/',
      element: <HomePage />,
      children: [
        {
          path: '/:id',
          element: <BookDescription />,
        },
      ],
      errorElement: (
        <NotFoundPage />
      ),
    },
  ],
);

const AppRouter = () => (
  <RouterProvider router={router} />
);

export default AppRouter;
