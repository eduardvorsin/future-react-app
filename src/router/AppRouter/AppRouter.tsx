import React from 'react';
import {
  createBrowserRouter,
  redirect,
  RouterProvider,
} from 'react-router-dom';
import BookDescription from '../../components/BL/BookDescription/BookDescription';
import HomePage from '../pages/HomePage/HomePage';
import NotFoundPage from '../pages/NotFoundPage/NotFoundPage';
import bookDescriptionLoader from '../loaders/bookDescriptionLoader';
import RouteError from '../../components/UI/RouteError/RouteError';

export type bookDescriptionPathName = '/:id';
const basename = process.env.NODE_ENV === 'production' ? '/future-react-app' : '';

export const routes = [
  {
    path: '/',
    loader: async () => redirect('/books'),
    errorElement: <NotFoundPage />,
  },
  {
    path: '/books',
    element: <HomePage />,
    children: [
      {
        path: ':id',
        element: <BookDescription />,
        errorElement: <RouteError />,
        loader: bookDescriptionLoader,
      },
    ],
  },
];

const router = createBrowserRouter(
  routes,
  { basename },
);

const AppRouter = () => (
  <RouterProvider router={router} />
);

export default AppRouter;
