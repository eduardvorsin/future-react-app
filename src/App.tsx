import React, { useCallback, useEffect, useMemo, useState } from 'react';
import AppRouter from './router/AppRouter/AppRouter';
import { ThemeContext, defaultTheme } from './contexts/ThemeContext';

export default function App() {
  const [theme, setTheme] = useState<'light' | 'dark'>(defaultTheme);
  const toggleTheme = useCallback(() => {
    setTheme((prevTheme) => (prevTheme === 'light' ? 'dark' : 'light'));
  }, [theme]);

  const themeContextValue = useMemo(() => ({
    value: theme,
    toggleTheme,
  }), [theme, toggleTheme]);

  useEffect(() => {
    document.documentElement.setAttribute('data-theme', theme);
    return () => {
      document.documentElement.removeAttribute('data-theme');
    };
  }, [theme]);

  return (
    <ThemeContext.Provider value={themeContextValue}>
      <div className='app'>
        <AppRouter />
      </div>

    </ThemeContext.Provider>
  );
}
