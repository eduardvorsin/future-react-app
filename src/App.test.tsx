import React from 'react';
import { render, screen, waitFor, waitForElementToBeRemoved } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import App from './App';
import ReduxWrapper from './tests/helpers/ReduxWrapper';
import i18n from './localization/i18next';
import dummyBooks from './tests/dummyBooks';
import { setupServerWithErrors } from './tests/server';
import { routes } from './router/AppRouter/AppRouter';

const router = createMemoryRouter(routes, { initialEntries: ['/'] });

describe('App tests', () => {
  afterEach(async () => {
    await waitFor(() => {
      i18n.changeLanguage('en');
    });
  });

  it('is rendered correctly', () => {
    render(<App />, { wrapper: ReduxWrapper });

    expect(screen.getByLabelText<HTMLLabelElement>(/language selection/i)).toBeInTheDocument();
    expect(screen.getByText<HTMLLegendElement>(/theme switcher/i)).toBeInTheDocument();
    expect(screen.getByRole<HTMLHeadingElement>('heading')).toBeInTheDocument();
    expect(screen.getByLabelText<HTMLLabelElement>(/categories/i)).toBeInTheDocument();
  });

  it('if you select another option in the language selector, the active language should change', async () => {
    const user = userEvent.setup();
    render(<App />, { wrapper: ReduxWrapper });

    await waitFor(async () => {
      await user.selectOptions(screen.getAllByRole<HTMLSelectElement>('combobox')[0], 'ru');
    });

    expect(screen.getByLabelText<HTMLLabelElement>(/выбор языка/i)).toBeInTheDocument();
    expect(screen.getByText<HTMLLegendElement>(/переключатель/i)).toBeInTheDocument();
    expect(screen.getByText<HTMLHeadingElement>(/поиск книг/i)).toBeInTheDocument();
    expect(screen.getByLabelText<HTMLLabelElement>(/категории/i)).toBeInTheDocument();
  });

  it('when switching the dark theme, the date attribute should be added to the html element', async () => {
    const user = userEvent.setup();
    render(<App />, { wrapper: ReduxWrapper });

    await waitFor(async () => {
      await user.click(screen.getAllByRole<HTMLInputElement>('switch')[1]);
    });

    expect(document.documentElement.dataset.theme).toBe('dark');
  });

  it('when you enter a book into the search and click on the search button, the list of books should load', async () => {
    const user = userEvent.setup();
    render(<App />, { wrapper: ReduxWrapper });

    await waitFor(async () => {
      await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'js');
    });
    await waitFor(async () => {
      await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /find/i }));
    });

    expect(screen.getByTitle<HTMLDivElement>('loading spinner')).toBeInTheDocument();

    await waitForElementToBeRemoved(screen.queryByTitle<HTMLDivElement>('loading spinner'));

    const bookItemTitle = new RegExp(dummyBooks.items[0].volumeInfo.title);
    const bookItemAuthors = new RegExp(dummyBooks.items[0].volumeInfo.authors[0]);
    const bookItemPrice = 'KZT 2,434';

    expect(screen.getByText<HTMLSpanElement>(bookItemPrice)).toBeInTheDocument();
    expect(screen.getByText<HTMLParagraphElement>(bookItemAuthors)).toBeInTheDocument();
    expect(screen.getByText<HTMLAnchorElement>(bookItemTitle)).toBeInTheDocument();
  });

  it('when you click on the learn more button, the url should change and render the page of a specific book', async () => {
    const user = userEvent.setup();
    const booksCount = i18n.t('booksCount', {
      count: 2,
      currentBooksCount: 2,
      ns: 'homePage',
    });

    render(<App />, { wrapper: ReduxWrapper });

    await waitFor(async () => {
      await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'js');
    });
    await waitFor(async () => {
      await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /find/i }));
    });
    await waitFor(async () => {
      await user.click(screen.getAllByRole<HTMLAnchorElement>('link', { name: /learn more/i })[0]);
    });

    await waitFor(async () => {
      expect(screen.queryByText<HTMLParagraphElement>(booksCount)).not.toBeInTheDocument();
    });

    expect(screen.getByText<HTMLSpanElement>(/Page count/i)).toBeInTheDocument();
    expect(screen.getByText<HTMLSpanElement>(/Book Type/i)).toBeInTheDocument();
    expect(screen.getByText<HTMLSpanElement>(/Published date/i)).toBeInTheDocument();
  });

  it('if there was some error with loading books, it should show it', async () => {
    setupServerWithErrors();
    const user = userEvent.setup();

    render(<App />, { wrapper: ReduxWrapper });

    await waitFor(async () => {
      await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'js');
    });
    await waitFor(async () => {
      await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /find/i }));
    });

    await waitForElementToBeRemoved(screen.queryByTitle<HTMLDivElement>('loading spinner'));

    expect(screen.queryByRole<HTMLAnchorElement>('link', { name: /learn more/i })).not.toBeInTheDocument();
    expect(screen.getByText<HTMLHeadingElement>(/failed to load/i)).toBeInTheDocument();
  });

  it('should render not found page if there is no corresponding route for the specified url', async () => {
    render(
      (
        <RouterProvider router={router} />
      ), { wrapper: ReduxWrapper },
    );

    await waitFor(() => {
      router.navigate('/about');
    });

    expect(screen.queryByLabelText<HTMLLabelElement>(/language selection/i)).not.toBeInTheDocument();
    expect(screen.queryByText<HTMLLegendElement>(/theme switcher/i)).not.toBeInTheDocument();
    expect(screen.queryByLabelText<HTMLLabelElement>(/categories/i)).not.toBeInTheDocument();

    expect(screen.getByText<HTMLHeadingElement>(/page not found/i)).toBeInTheDocument();
    expect(screen.getByText<HTMLAnchorElement>(/back to main page/i)).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(<App />, { wrapper: ReduxWrapper });

    expect(document.body).toMatchSnapshot();
  });

  it('is a snapshot with a list of books', async () => {
    const user = userEvent.setup();
    render(<App />, { wrapper: ReduxWrapper });

    await waitFor(async () => {
      await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'js');
    });
    await waitFor(async () => {
      await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /find/i }));
    });

    expect(screen.getByTitle<HTMLDivElement>('loading spinner')).toBeInTheDocument();

    await waitForElementToBeRemoved(screen.queryByTitle<HTMLDivElement>('loading spinner'));

    expect(document.body).toMatchSnapshot();
  });

  it('is a snapshot with a fetch error message', async () => {
    setupServerWithErrors();
    const user = userEvent.setup();

    render(<App />, { wrapper: ReduxWrapper });

    await waitFor(async () => {
      await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'js');
    });
    await waitFor(async () => {
      await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /find/i }));
    });

    await waitForElementToBeRemoved(screen.queryByTitle<HTMLDivElement>('loading spinner'));

    expect(document.body).toMatchSnapshot();
  });
});
