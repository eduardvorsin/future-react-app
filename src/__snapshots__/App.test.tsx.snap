// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests is a basic snapshot 1`] = `
<body>
  <div>
    <div
      class="home-page"
    >
      <div
        class="home-page__quick-menu"
      >
        <div
          class="home-page__quick-menu-item"
        >
          <fieldset
            class="switcher theme-switcher undefined"
          >
            <legend
              class="switcher__label sr-only"
            >
              Color theme switcher
            </legend>
            <label
              class="switcher__input-label sr-only"
              for=":r7:-input-first"
            >
              светлая тема
            </label>
            <input
              aria-checked="false"
              class="switcher__radio"
              id=":r7:-input-first"
              name="theme-switcher"
              role="switch"
              type="radio"
              value="light"
            />
            <label
              class="switcher__input-label sr-only"
              for=":r7:-input-second"
            >
              темная тема
            </label>
            <input
              aria-checked="true"
              checked=""
              class="switcher__radio"
              id=":r7:-input-second"
              name="theme-switcher"
              role="switch"
              type="radio"
              value="dark"
            />
            <span
              class="switcher__status"
            />
          </fieldset>
        </div>
        <div
          class="home-page__quick-menu-item"
        >
          <div
            class="language__select undefined select"
          >
            <label
              class="select__label"
              for="language-select"
            >
              Language selection
            </label>
            <div
              class="select-wrapper"
            >
              <select
                class="select-wrapper__select"
                id="language-select"
              >
                <option
                  value="en"
                >
                  ENG
                </option>
                <option
                  value="ru"
                >
                  RUS
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <section
        class="home-page__hero hero-section"
      >
        <div
          class="hero-section__container container"
        >
          <h2
            class="title hero-section__title title--dark title--level-2"
          >
            Book Search
          </h2>
          <div
            class="books-search hero-section__content"
          >
            <form
              class="books-search__search-bar search-form search-form__search-bar"
            >
              <div
                class="search-bar"
                role="search"
              >
                <label
                  class="search-bar__label sr-only"
                  for="search-bar"
                >
                  Search Bar
                </label>
                <input
                  class="search-bar__input"
                  id="search-bar"
                  placeholder="Find a book..."
                  value=""
                />
                <button
                  class="button--icon-button search-bar__clear"
                >
                  <svg />
                  Clear
                </button>
                <button
                  class="button--icon-button search-bar__find"
                  type="submit"
                >
                  <svg />
                  To find
                </button>
              </div>
            </form>
            <div
              class="books-search__category-select books-search__select select"
            >
              <label
                class="select__label"
                for="categories"
              >
                Categories
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="categories"
                >
                  <option
                    value="all"
                  >
                    All
                  </option>
                  <option
                    value="art"
                  >
                    Art
                  </option>
                  <option
                    value="biography"
                  >
                    Biography
                  </option>
                  <option
                    value="computers"
                  >
                    Computers
                  </option>
                  <option
                    value="history"
                  >
                    History
                  </option>
                  <option
                    value="medical"
                  >
                    Medical
                  </option>
                  <option
                    value="poetry"
                  >
                    Poetry
                  </option>
                </select>
              </div>
            </div>
            <div
              class="books-search__sort-select books-search__select select"
            >
              <label
                class="select__label"
                for="sortingBy"
              >
                Sort by
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="sortingBy"
                >
                  <option
                    value="relevance"
                  >
                    Relevance
                  </option>
                  <option
                    value="newest"
                  >
                    Newest
                  </option>
                </select>
              </div>
            </div>
            <div
              class="books-search__find-by-select books-search__select select"
            >
              <label
                class="select__label"
                for="findBy"
              >
                Search by
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="findBy"
                >
                  <option
                    value="intitle"
                  >
                    By title
                  </option>
                  <option
                    value="inauthor"
                  >
                    By author
                  </option>
                  <option
                    value="inpublisher"
                  >
                    By publisher
                  </option>
                  <option
                    value="isbn"
                  >
                    By ISBN
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="books-section"
      >
        <div
          class="books-section__container container "
        />
      </section>
    </div>
  </div>
</body>
`;

exports[`App tests is a snapshot with a fetch error message 1`] = `
<body>
  <div>
    <div
      class="home-page"
    >
      <div
        class="home-page__quick-menu"
      >
        <div
          class="home-page__quick-menu-item"
        >
          <fieldset
            class="switcher theme-switcher undefined"
          >
            <legend
              class="switcher__label sr-only"
            >
              Color theme switcher
            </legend>
            <label
              class="switcher__input-label sr-only"
              for=":r9:-input-first"
            >
              светлая тема
            </label>
            <input
              aria-checked="false"
              class="switcher__radio"
              id=":r9:-input-first"
              name="theme-switcher"
              role="switch"
              type="radio"
              value="light"
            />
            <label
              class="switcher__input-label sr-only"
              for=":r9:-input-second"
            >
              темная тема
            </label>
            <input
              aria-checked="true"
              checked=""
              class="switcher__radio"
              id=":r9:-input-second"
              name="theme-switcher"
              role="switch"
              type="radio"
              value="dark"
            />
            <span
              class="switcher__status"
            />
          </fieldset>
        </div>
        <div
          class="home-page__quick-menu-item"
        >
          <div
            class="language__select undefined select"
          >
            <label
              class="select__label"
              for="language-select"
            >
              Language selection
            </label>
            <div
              class="select-wrapper"
            >
              <select
                class="select-wrapper__select"
                id="language-select"
              >
                <option
                  value="en"
                >
                  ENG
                </option>
                <option
                  value="ru"
                >
                  RUS
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <section
        class="home-page__hero hero-section"
      >
        <div
          class="hero-section__container container"
        >
          <h2
            class="title hero-section__title title--dark title--level-2"
          >
            Book Search
          </h2>
          <div
            class="books-search hero-section__content"
          >
            <form
              class="books-search__search-bar search-form search-form__search-bar"
            >
              <div
                class="search-bar"
                role="search"
              >
                <label
                  class="search-bar__label sr-only"
                  for="search-bar"
                >
                  Search Bar
                </label>
                <input
                  class="search-bar__input"
                  id="search-bar"
                  placeholder="Find a book..."
                  value="js"
                />
                <button
                  class="button--icon-button search-bar__clear"
                >
                  <svg />
                  Clear
                </button>
                <button
                  class="button--icon-button search-bar__find"
                  type="submit"
                >
                  <svg />
                  To find
                </button>
              </div>
            </form>
            <div
              class="books-search__category-select books-search__select select"
            >
              <label
                class="select__label"
                for="categories"
              >
                Categories
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="categories"
                >
                  <option
                    value="all"
                  >
                    All
                  </option>
                  <option
                    value="art"
                  >
                    Art
                  </option>
                  <option
                    value="biography"
                  >
                    Biography
                  </option>
                  <option
                    value="computers"
                  >
                    Computers
                  </option>
                  <option
                    value="history"
                  >
                    History
                  </option>
                  <option
                    value="medical"
                  >
                    Medical
                  </option>
                  <option
                    value="poetry"
                  >
                    Poetry
                  </option>
                </select>
              </div>
            </div>
            <div
              class="books-search__sort-select books-search__select select"
            >
              <label
                class="select__label"
                for="sortingBy"
              >
                Sort by
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="sortingBy"
                >
                  <option
                    value="relevance"
                  >
                    Relevance
                  </option>
                  <option
                    value="newest"
                  >
                    Newest
                  </option>
                </select>
              </div>
            </div>
            <div
              class="books-search__find-by-select books-search__select select"
            >
              <label
                class="select__label"
                for="findBy"
              >
                Search by
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="findBy"
                >
                  <option
                    value="intitle"
                  >
                    By title
                  </option>
                  <option
                    value="inauthor"
                  >
                    By author
                  </option>
                  <option
                    value="inpublisher"
                  >
                    By publisher
                  </option>
                  <option
                    value="isbn"
                  >
                    By ISBN
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="books-section"
      >
        <div
          class="books-section__container container "
        >
          <div
            class="books-section__error error "
            role="alert"
          >
            <h4
              class="title error__title title--dark title--level-5"
            >
              Failed to load books, try again
            </h4>
          </div>
        </div>
      </section>
    </div>
  </div>
</body>
`;

exports[`App tests is a snapshot with a list of books 1`] = `
<body>
  <div>
    <div
      class="home-page"
    >
      <div
        class="home-page__quick-menu"
      >
        <div
          class="home-page__quick-menu-item"
        >
          <fieldset
            class="switcher theme-switcher undefined"
          >
            <legend
              class="switcher__label sr-only"
            >
              Color theme switcher
            </legend>
            <label
              class="switcher__input-label sr-only"
              for=":r8:-input-first"
            >
              светлая тема
            </label>
            <input
              aria-checked="false"
              class="switcher__radio"
              id=":r8:-input-first"
              name="theme-switcher"
              role="switch"
              type="radio"
              value="light"
            />
            <label
              class="switcher__input-label sr-only"
              for=":r8:-input-second"
            >
              темная тема
            </label>
            <input
              aria-checked="true"
              checked=""
              class="switcher__radio"
              id=":r8:-input-second"
              name="theme-switcher"
              role="switch"
              type="radio"
              value="dark"
            />
            <span
              class="switcher__status"
            />
          </fieldset>
        </div>
        <div
          class="home-page__quick-menu-item"
        >
          <div
            class="language__select undefined select"
          >
            <label
              class="select__label"
              for="language-select"
            >
              Language selection
            </label>
            <div
              class="select-wrapper"
            >
              <select
                class="select-wrapper__select"
                id="language-select"
              >
                <option
                  value="en"
                >
                  ENG
                </option>
                <option
                  value="ru"
                >
                  RUS
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <section
        class="home-page__hero hero-section"
      >
        <div
          class="hero-section__container container"
        >
          <h2
            class="title hero-section__title title--dark title--level-2"
          >
            Book Search
          </h2>
          <div
            class="books-search hero-section__content"
          >
            <form
              class="books-search__search-bar search-form search-form__search-bar"
            >
              <div
                class="search-bar"
                role="search"
              >
                <label
                  class="search-bar__label sr-only"
                  for="search-bar"
                >
                  Search Bar
                </label>
                <input
                  class="search-bar__input"
                  id="search-bar"
                  placeholder="Find a book..."
                  value="js"
                />
                <button
                  class="button--icon-button search-bar__clear"
                >
                  <svg />
                  Clear
                </button>
                <button
                  class="button--icon-button search-bar__find"
                  type="submit"
                >
                  <svg />
                  To find
                </button>
              </div>
            </form>
            <div
              class="books-search__category-select books-search__select select"
            >
              <label
                class="select__label"
                for="categories"
              >
                Categories
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="categories"
                >
                  <option
                    value="all"
                  >
                    All
                  </option>
                  <option
                    value="art"
                  >
                    Art
                  </option>
                  <option
                    value="biography"
                  >
                    Biography
                  </option>
                  <option
                    value="computers"
                  >
                    Computers
                  </option>
                  <option
                    value="history"
                  >
                    History
                  </option>
                  <option
                    value="medical"
                  >
                    Medical
                  </option>
                  <option
                    value="poetry"
                  >
                    Poetry
                  </option>
                </select>
              </div>
            </div>
            <div
              class="books-search__sort-select books-search__select select"
            >
              <label
                class="select__label"
                for="sortingBy"
              >
                Sort by
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="sortingBy"
                >
                  <option
                    value="relevance"
                  >
                    Relevance
                  </option>
                  <option
                    value="newest"
                  >
                    Newest
                  </option>
                </select>
              </div>
            </div>
            <div
              class="books-search__find-by-select books-search__select select"
            >
              <label
                class="select__label"
                for="findBy"
              >
                Search by
              </label>
              <div
                class="select-wrapper"
              >
                <select
                  class="select-wrapper__select"
                  id="findBy"
                >
                  <option
                    value="intitle"
                  >
                    By title
                  </option>
                  <option
                    value="inauthor"
                  >
                    By author
                  </option>
                  <option
                    value="inpublisher"
                  >
                    By publisher
                  </option>
                  <option
                    value="isbn"
                  >
                    By ISBN
                  </option>
                </select>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="books-section"
      >
        <div
          class="books-section__container container "
        >
          <p
            class="0 books-count"
          >
            Presented 2 of (2) books
          </p>
          <div
            class=" books-list"
          >
            <article
              class="books-list__item card"
            >
              <div
                class="card__img-wrapper "
                data-testid="image-wrapper"
              >
                <img
                  alt="{Вы не знаете JS} Замыкания и объекты book cover"
                  class="card__img"
                  height="185"
                  src="http://books.google.com/books/content?id=6_-eDwAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
                  width="128"
                />
                <a
                  href="/books/6_-eDwAAQBAJ"
                >
                  Learn more
                </a>
              </div>
              <div
                class="card__content"
              >
                <p
                  class="card__price"
                >
                  <span
                    class="sr-only"
                  >
                    Book price
                    :
                  </span>
                  KZT 2,434
                </p>
                <h3
                  class="title card__title title--dark title--level-6"
                >
                  <a
                    href="/books/6_-eDwAAQBAJ"
                  >
                    {Вы не знаете JS} Замыкания и объекты
                  </a>
                </h3>
                <p
                  class="card__authors"
                >
                  Authors
                  : 
                  Симпсон Кайл
                </p>
              </div>
              <a
                aria-disabled="false"
                class="link card__more-link  link--secondary"
                href="/books/6_-eDwAAQBAJ"
              >
                Learn more
              </a>
            </article>
            <article
              class="books-list__item card"
            >
              <div
                class="card__img-wrapper "
                data-testid="image-wrapper"
              >
                <img
                  alt="Мой, и только мой book cover"
                  class="card__img"
                  height="185"
                  src="test-file-stub"
                  width="128"
                />
                <a
                  href="/books/Js-tOgAACAAJ"
                >
                  Learn more
                </a>
              </div>
              <div
                class="card__content"
              >
                <p
                  class="card__price"
                >
                  <span
                    class="sr-only"
                  >
                    Book price
                    :
                  </span>
                  Not available
                </p>
                <h3
                  class="title card__title title--dark title--level-6"
                >
                  <a
                    href="/books/Js-tOgAACAAJ"
                  >
                    Мой, и только мой
                  </a>
                </h3>
                <p
                  class="card__authors"
                >
                  Authors
                  : 
                  Сьюзен Элизабет Филлипс
                </p>
              </div>
              <a
                aria-disabled="false"
                class="link card__more-link  link--secondary"
                href="/books/Js-tOgAACAAJ"
              >
                Learn more
              </a>
            </article>
          </div>
        </div>
      </section>
    </div>
  </div>
</body>
`;
